src/runtime/server/middleware/cspNonceHandler.ts

```diff
@@ -0,0 +1,46 @@
+import crypto from 'node:crypto'
+import { createError, defineEventHandler, getCookie, sendError, setCookie } from 'h3'
+import { getRouteRules } from '#imports'
+
+export type NonceOptions = {
+  enabled: boolean;
+  mode: 'renew' | 'check';
+  value: undefined | (() => string);
+}
+
+export default defineEventHandler((event) => {
+  let csp = `${event.node.res.getHeader('Content-Security-Policy')}`
+  const routeRules = getRouteRules(event)
+
+  if (routeRules.security.nonce !== false) {
+    const nonceConfig: NonceOptions = routeRules.security.nonce
+
+    // See if we are checking the nonce against the current value, or if we are renewing the nonce value
+    let nonce: string | undefined
+    switch (nonceConfig?.mode) {
+      case 'check': {
+        nonce = getCookie(event, 'nonce')
```

Comment:
Is it secured to use cookie for nonce value? I do not know, sorry, but what do you think about https://nuxt.com/docs/api/composables/use-nuxt-data#usenuxtdata , for example?

Reply:
As far as I know, using a cookie with `sameSite` and `secure` will only allow access to that cookie from scripts which are already loaded, i.e. which adhere to the nonce value.

I tried using useNuxtData and similar, but there seems no added benefit to security, and the implementation becomes more complex as you need to add specific endpoints for retrieving the nonce value (and how is it going to persist between regeneration?)

Furthermore, `@sidebase/nuxt-auth` uses a similar approach to pass tokens to the browser as well. It's also recommended by the nuxt documentation.

I'll see if I can find the correct links for this.

Reply:
I have not think about non ssr variant... Now cookies is looked like very good solution))